Sources/Imaging/Microsoft.Psi.Imaging.Windows/DepthImageFromTiffStreamDecoder.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.Psi.Imaging
+{
+    using System.IO;
+    using System.Windows;
+    using System.Windows.Media.Imaging;
+
+    /// <summary>
+    /// Implements a depth image decoder.
+    /// </summary>
+    public class DepthImageFromTiffStreamDecoder : IDepthImageFromStreamDecoder
```

Comment:
I'm not sure you need to define a TIFF-specific decoder class. If you look at the `DepthImageFromStreamDecoder` class and how that is used, it should be capable of instantiating the correct decoder (e.g. PNG, TIFF) based on the stream. Did you try it (in your test method) and if so did it not work for decoding your TIFF-encoded depth images?

Reply:
You are right, the existing decoder still works for TIFF files. I did not try it before and rolling back the change now.